lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -1637,6 +1637,65 @@ VTableLoadMethodOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return success();
 }
 
+LogicalResult VTableOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  Operation *self = getOperation();
+
+  // sym_name must end with ::@vtable and root must be a ClassDeclOp
+  SymbolRefAttr name = getSymNameAttr();
+  if (!name)
+    return emitOpError("requires 'sym_name' SymbolRefAttr");
+
+  // Last component must be "vtable".
+  if (name.getNestedReferences().empty() ||
+      name.getNestedReferences().back().getValue() != "vtable")
+    return emitOpError() << "sym_name must end with '::@vtable', got " << name;
+
+  // Root symbol must resolve (from the nearest symbol table) to a ClassDeclOp.
+  Operation *rootDef = symbolTable.lookupNearestSymbolFrom(
+      self, SymbolRefAttr::get(name.getRootReference()));
+  if (!rootDef)
+    return emitOpError() << "cannot resolve root class symbol '"
+                         << name.getRootReference() << "' for sym_name "
+                         << name;
+
+  if (!isa<ClassDeclOp>(rootDef))
+    return emitOpError()
+           << "root of sym_name must name a 'moore.class.classdecl', got "
+           << name;
+
+  Block &block = getBody().front();
+
+  // Ensure only allowed ops appear inside.
+  for (Operation &op : block) {
+    if (!isa<VTableOp, VTableEntryOp>(op))
+      return emitOpError(
+          "body may only contain 'moore.vtable' or 'moore.vtable_entry' ops");
+  }
```

Comment:
Can you move this into a region verifier, since it doesn't really involve verifying the symbol use but rather the body region of the op?

Reply:
Done!